### DISCUSSION TITLE
Platform agnostic forms

### DISCUSSION THREAD

**Discussion by timvandam** [edmundhung/conform#749] [Ideas]
Hi! I love the DX of conform with remix, and would like to use it outside of web-based React. However, I am currently working in two different environments where working work conform is near-impossible because they are not web/html-based or because I am must use form components that do not expose fields like `id`. I think the main bottleneck here is the fact that there is no `id` field, which conform relies on. Would you be open to the idea of making conform platform-agnostic? For instance by using react context instead of form ids. Aside from this I think there is not much else that would be required - equivalents for `getFormProps` in other languages would be easy for users to make themselves etc

**Comment by edmundhung** [comment]
It is not impossible but it would be quite hard to justify whether the complexity to re-implement HTML form and the corresponding DOM APIs worth the effort. This will includes:

- Associating form elements based on the hierarchy or form `id` (e.g. `form.elements` includes references to all input elements. You can also find the corresponding form element through `inputElement.form`)
- Ability to trigger a form submission programmatically (e.g. `inputElement.form.requestSubmit(submitter)`)
- Ability to modify a input element programmatically (e.g. `inputElement.value = 'value to be updated'`)
- Construct the form value based on the hierarchy of all input elements just by referring to the form (e.g `new FormData(formElement)`)
- Respect the submitter (button that trigger the submission) and adjust the form data based on its `name` and `value` (for the intent buttons)

It would be fun to explore this one day but it's unlikely a priority for me until the project is at a more mature stage.

**Comment by MattyBalaam** [comment]
As a companion to this. Are there any guidelines around how you could integrate conform with web components? I started to explore them and how they dffer from traditional framework ones and started finding there were maybe quite a few hoops to jump through.
